### PR TITLE
chore: Generate docs as a precommit hook

### PR DESCRIPTION
**Pull request by lukehesluke**
So that we don't have to remember to do this for new tests

## Demonstration

Screenshot 2020-07-28 at 17 44 25

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,5 +18,13 @@
   },
   "engines": {
     "node": "12.18.2"
+  },
+  "devDependencies": {
+    "husky": "^4.2.5"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "npm run doc-gen --prefix packages/openactive-integration-tests"
```
Comment (nickevansuk) [inline package.json:27]: So it turns out that "--prefix packages/openactive-integration-tests" doesn't run on Windows, which hinders development on Windows if this is in a hook. Suggest replacing this with a `cd` prefix similar to https://github.com/openactive/openactive-test-suite/pull/116

Comment (lukehesluke) [reply package.json:27]: @nickevansuk is it just `npm install` which doesn't work with `--prefix` on windows? e.g. the only [bug](https://github.com/npm/cli/issues/1290) [reports](https://html.developreference.com/article/11036084/NPM+prefix+flag+does+not+work+as+expected+on+windows) [I can find](https://github.com/npm/npm/issues/20245) relate to `npm install` and I see that we've only removed `--prefix` from our `install` script. Our test script still looks like `npm test --prefix packages/openactive-broker-microservice ...`

Comment (lukehesluke) [reply package.json:27]: My apologies, I have now seen and commented on your other PR (https://github.com/openactive/openactive-test-suite/pull/116), which changes all of the scripts

Comment (nickevansuk) [reply package.json:27]: No worries - based on my testing it is mainly `install` that has the issue, but it appears to be an anti-pattern more generally? - was looking for a reference in the npm docs or repos where `--prefix` is recommended for this purpose and couldn't find one... 

If we choose to use `cd` here, would be good to check that husky uses a sub-shell like `npm` does (see https://github.com/openactive/openactive-test-suite/pull/116#issuecomment-665567302), to ensure there's no issue with the working dir restore.

If not perhaps we add an npm script in the root that calls `npm run doc-gen` in the packages dir using `cd`, then just call this from `husky`? Perhaps better taking this approach anyway for consistency?

Comment (lukehesluke) [reply package.json:27]: > If not perhaps we add an npm script in the root that calls npm run doc-gen in the packages dir using cd, then just call this from husky? Perhaps better taking this approach anyway for consistency?

👍  good idea

Comment (lukehesluke) [reply package.json]: @nickevansuk this is now done

**Pull request merged**